### PR TITLE
Detect and fix keyerror

### LINKED ISSUE #40: KeyError "type" in connection
When a connection is to be removed ```connection['type']``` causes a KeyError when using the initscripts backend and no type is defined. This should fix it:

```
diff --git a/library/network_connections.py b/library/network_connections.py
index 271ab40..4922b13 100755
--- a/library/network_connections.py
+++ b/library/network_connections.py
@@ -2650,7 +2650,7 @@ class Cmd_initscripts(Cmd):
     def run_prepare(self):
         Cmd.run_prepare(self)
         for idx, connection in enumerate(self.connections):
-            if connection['type'] in [ 'macvlan' ]:
+            if connection.get('type') in [ 'macvlan' ]:
                 self.log_fatal(idx, 'unsupported type %s for initscripts provider' % (connection['type']))
 
     def check_name(self, idx, name = None):
```

The exception was: ```File "/tmp/ansible_jD4yqb/ansible_module_network_connections.py", line 2813, in <module>     cmd.run()   File
"/tmp/ansible_jD4yqb/ansible_module_network_connections.py", line 2398, in run     self.run_prepare()   File "/tmp/ansible_jD4yqb/ansible_module_network_connections.py", line 2654, in run_prepare     if
connection['type'] in [ 'macvlan' ]: KeyError: 'type'```

### PR DESCRIPTION
**Pull request by tyll**
This demonstrates and fixes #40 

It also contains commits from #50 that should be merged there but are required to create the test case here

### CODE REVIEW

File: library/network_connections.py [hunk 1]
```diff
@@ -2650,7 +2650,7 @@ def __init__(self, **kwargs):
     def run_prepare(self):
         Cmd.run_prepare(self)
         for idx, connection in enumerate(self.connections):
-            if connection['type'] in [ 'macvlan' ]:
+            if connection.get('type') in [ 'macvlan' ]:
```
Comment (thom311) [inline library/network_connections.py]: shouldn't it be `connection.get('type', None)` ?

Comment (tyll) [reply library/network_connections.py]: The default value is `None`.

Comment (thom311) [reply library/network_connections.py]: A, I didn't realize. I thought by default it throws an exception, that's why I often explicitly set `None` as default value.

Comment (tyll) [reply library/network_connections.py]: The exception happens when one uses `[]` instead of `.get()`.

**Pull request merged**